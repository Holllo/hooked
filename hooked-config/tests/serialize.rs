use {
  hooked_config::{Config, ExitAction, PreCommit, Task},
  toml::to_string_pretty,
};

use insta::assert_snapshot;

#[test]
fn test_serialize() {
  let pre_commit_command = PreCommit {
    name: Some("Command Test".to_string()),
    on_failure: ExitAction::Continue,
    task: Task {
      command: Some("exit 0".to_string()),
      script: None,
    },
  };

  let pre_commit_script = PreCommit {
    name: Some("Script Test".to_string()),
    on_failure: ExitAction::Stop,
    task: Task {
      command: None,
      script: Some("test.sh".into()),
    },
  };

  let config = Config {
    general: Default::default(),
    pre_commit: vec![pre_commit_command, pre_commit_script],
  };

  assert_snapshot!("serialize", to_string_pretty(&config).unwrap());
}
