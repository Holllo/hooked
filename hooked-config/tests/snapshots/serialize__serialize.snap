---
source: hooked-config/tests/serialize.rs
expression: to_string_pretty(&config).unwrap()
---
[general]
config = 'Hooked.toml'
directory = 'hooks'

[[pre_commit]]
name = 'Command Test'
on_failure = 'continue'
staged = ['*.txt']
command = 'exit 0'

[[pre_commit]]
name = 'Script Test'
on_failure = 'stop'
staged = []
script = 'test.sh'

